evm-node.md

```diff
@@ -0,0 +1,149 @@
+    syntax Bool ::= #isCodeEmpty(Int) [function, hook(MANTIS.isCodeEmpty)]
+                  | #accountExists(Int) [function, hook(MANTIS.accountExists)]
+ // --------------------------------------------------------------------------
+    rule <k> #loadAccount ACCT => . ... </k>
+         <activeAccounts> ACCTS (.Set => SetItem(ACCT)) </activeAccounts>
+         <accounts>
+           ( .Bag
+          => <account>
+               <acctID> ACCT </acctID>
+               <balance> #getBalance(ACCT) </balance>
+               <code> #if #isCodeEmpty(ACCT) #then .WordStack #else #unloaded #fi </code>
+               <storage> .Map </storage>
+               <nonce> #getNonce(ACCT) </nonce>
+             </account>
+           )
+           ...
+         </accounts>
+      requires notBool ACCT in ACCTS andBool #accountExists(ACCT)
+
+    rule <k> #loadAccount ACCT => . ... </k>
+         <activeAccounts> ACCTS </activeAccounts>
+      requires ACCT in ACCTS orBool notBool #accountExists(ACCT)
+
+    syntax Int ::= #getStorageData(Int, Int) [function, hook(MANTIS.getStorageData)]
+ // --------------------------------------------------------------------------------
+    rule <k> #lookupStorage ACCT INDEX => . ... </k>
+         <account>
+           <acctID>  ACCT                                                         </acctID>
+           <storage> STORAGE => STORAGE [ INDEX <- #getStorageData(ACCT, INDEX) ] </storage>
+           ...
+         </account>
+      requires notBool INDEX in_keys(STORAGE)
+
+    rule <k> #lookupStorage ACCT INDEX => . ... </k>
+         <account>
+           <acctID> ACCT </acctID>
+           <storage> ... INDEX |-> _ ... </storage>
+           ...
+         </account>
+
+    syntax String ::= #getCode(Int) [function, hook(MANTIS.getCode)]
+ // ----------------------------------------------------------------
+    rule <k> #lookupCode ACCT => . ... </k>
+         <account>
+           <acctID> ACCT </acctID>
+           <code> #unloaded => #parseByteStackRaw(#getCode(ACCT)) </code>
+           ...
+         </account>
+
+    rule <k> #lookupCode ACCT => . ... </k>
+         <account>
+           <acctID> ACCT </acctID>
+           <code> _:WordStack </code>
+           ...
+         </account>
+
+    rule #lookupCode ACCT => .
+      requires notBool #accountExists(ACCT)
+
+    syntax Int ::= #getBlockhash(Int) [function, hook(MANTIS.getBlockhash)]
+ // -----------------------------------------------------------------------
+    rule <k> BLOCKHASH N => #getBlockhash(N) ~> #push ... </k> <mode> NORMAL </mode>
+      requires N >=Int 0 andBool N <Int 256
+    rule <k> BLOCKHASH N => 0 ~> #push ... </k> <mode> NORMAL </mode>
+      requires N <Int 0 orBool N >=Int 256
+
+    syntax EthereumSimulation ::= runVM(iscreate: Bool, to: Int, from: Int, code: String, args: String, value: Int, gasprice: Int, gas: Int, beneficiary: Int, difficulty: Int, number: Int, gaslimit: Int, timestamp: Int, unused: String)
+ 
```

Comment:
I've never seen this syntax for K. Is it some sort of record syntax?

Just it just enable the line below (matching on a subset of arguments?)

Reply:
yes